pvnet/data/datamodule.py

```diff
@@ -1,57 +1,111 @@
 """ Data module for pytorch lightning """
-
-import resource
+from glob import glob
 
 import torch
-from ocf_datapipes.batch import batch_to_tensor, stack_np_examples_into_batch
-from ocf_datapipes.training.pvnet import pvnet_datapipe
-from torch.utils.data.datapipes.iter import FileLister
+from lightning.pytorch import LightningDataModule
+from ocf_data_sampler.torch_datasets.pvnet_uk_regional import PVNetUKRegionalDataset
+from ocf_datapipes.batch import NumpyBatch, batch_to_tensor, stack_np_examples_into_batch
+from torch.utils.data import DataLoader, Dataset
+
+
+class NumpybatchPremadeSamplesDataset(Dataset):
+    """Dataset to load NumpyBatch samples"""
+
+    def __init__(self, sample_dir):
+        """Dataset to load NumpyBatch samples
+
+        Args:
+            sample_dir: Path to the directory of pre-saved samples.
+        """
+        self.sample_paths = glob(f"{sample_dir}/*.pt")
 
-from pvnet.data.base import BaseDataModule
+    def __len__(self):
+        return len(self.sample_paths)
 
-rlimit = resource.getrlimit(resource.RLIMIT_NOFILE)
-resource.setrlimit(resource.RLIMIT_NOFILE, (2048, rlimit[1]))
+    def __getitem__(self, idx):
+        return torch.load(self.sample_paths[idx])
 
 
-class DataModule(BaseDataModule):
+def collate_fn(samples: list[NumpyBatch]):
```

Comment:
Small one but it might be nice to have a return type hint here:
```suggestion
def collate_fn(samples: list[NumpyBatch]) -> TensorBatch:
```
just so it's a bit clearer what this function returns, would require importing that type from ocf_datapipes and eventually we would put this or something similar into ocf-data-sampler to remove that dependency

Reply:
Yeh ,sounds good